python/cudf/cudf/api/types.py

```diff
@@ -230,13 +232,240 @@ def _union_categoricals(
+    elif isinstance(arr_or_dtype, cudf.Series):
+        return pd_types.is_object_dtype(arr_or_dtype=arr_or_dtype.dtype)
+    else:
+        return pd_types.is_object_dtype(arr_or_dtype=arr_or_dtype)
+
+
+def is_float_dtype(arr_or_dtype) -> bool:
+    """
+    Check whether the provided array or dtype is of a float dtype.
+
+    Parameters
+    ----------
+    arr_or_dtype : array-like or dtype
+        The array or dtype to check.
+
+    Returns
+    -------
+    boolean
+        Whether or not the array or dtype is of a float dtype.
+
+    Examples
+    --------
+    >>> from cudf.api.types import is_float_dtype
+    >>> import numpy as np
+    >>> import cudf
+    >>> is_float_dtype(str)
+    False
+    >>> is_float_dtype(int)
+    False
+    >>> is_float_dtype(float)
+    True
+    >>> is_float_dtype(np.array(['a', 'b']))
+    False
+    >>> is_float_dtype(cudf.Series([1, 2]))
+    False
+    >>> is_float_dtype(cudf.Index([1, 2.]))
+    True
+    """
+    if isinstance(arr_or_dtype, cudf.BaseIndex):
+        return arr_or_dtype._is_floating()
+    elif (
+        (isclass(arr_or_dtype) and issubclass(arr_or_dtype, _BaseDtype))
+        or isinstance(arr_or_dtype, _BaseDtype)
+        or isinstance(getattr(arr_or_dtype, "dtype", None), _BaseDtype)
+    ):
+        return False
+    return pd_types.is_float_dtype(arr_or_dtype=arr_or_dtype)
+
+
+def is_integer_dtype(arr_or_dtype) -> bool:
+    """
+    Check whether the provided array or dtype is of an integer dtype.
+    Unlike in `is_any_int_dtype`, timedelta64 instances will return False.
+
+    Parameters
+    ----------
+    arr_or_dtype : array-like or dtype
+        The array or dtype to check.
+
+    Returns
+    -------
+    boolean
+        Whether or not the array or dtype is of an integer dtype and
+        not an instance of timedelta64.
+
+    Examples
+    --------
+    >>> from cudf.api.types import is_integer_dtype
+    >>> import numpy as np
+    >>> import cudf
+    >>> is_integer_dtype(str)
+    False
+    >>> is_integer_dtype(int)
+    True
+    >>> is_integer_dtype(float)
+    False
+    >>> is_integer_dtype(np.uint64)
+    True
+    >>> is_integer_dtype('int8')
+    True
+    >>> is_integer_dtype('Int8')
+    True
+    >>> is_integer_dtype(np.datetime64)
+    False
+    >>> is_integer_dtype(np.timedelta64)
+    False
+    >>> is_integer_dtype(np.array(['a', 'b']))
+    False
+    >>> is_integer_dtype(cudf.Series([1, 2]))
+    True
+    >>> is_integer_dtype(np.array([], dtype=np.timedelta64))
+    False
+    >>> is_integer_dtype(cudf.Index([1, 2.]))  # float
+    False
+    """
+    if isinstance(arr_or_dtype, cudf.BaseIndex):
+        return arr_or_dtype._is_integer()
+    elif (
+        (isclass(arr_or_dtype) and issubclass(arr_or_dtype, _BaseDtype))
+        or isinstance(arr_or_dtype, _BaseDtype)
+        or isinstance(getattr(arr_or_dtype, "dtype", None), _BaseDtype)
+    ):
+        return False
+    return pd_types.is_integer_dtype(arr_or_dtype=arr_or_dtype)
```

Comment:
This is definitely something we want to avoid duplicating across all these functions. We can still reuse`_wrap_pandas_is_dtype_api` like this, but there might be a cleaner refactor (not a priority IMO):

```python
_wrap_pandas_is_dtype_api(pd.api.types.is_integer_dtype)(arr_or_dtype=arr_or_dtype)
```

Reply:
Yup, made this change 👍